tests/generative/variable_strategies.py

```diff
@@ -292,8 +290,11 @@ def many_rows_per_patient_frame(draw):
 
     @st.composite
     def sorted_frame(draw):
-        source = draw(many_rows_per_patient_frame())
-        return draw(sort(source))
+        source = draw(select_table())
+        # for _ in range(draw(st.integers(min_value=1, max_value=3))):
+        #     source = draw(sort(source))
+        source = draw(sort(source))
+        return source
```

Comment:
@benbc I started trying to update this PR as we discussed this morning.  However, if I add the commented out lines here or the ones in `sort()` below, I start to see the too-many-filters error.  I also started to see it when removed those lines, and I could get it to go away by reducing the max number of filters in `many_rows_per_patient_frame`.

Are the for loops in `many_rows_per_patient_frame` and `ancestor_of` discarding examples?

Reply:
No, they shouldn't be. I don't understand why that would be. It can't be using filtering to get integers that fall within the min/max bounds, can it?

Reply:
The IntegerStrategy does do something with filtering, but I'm not sure I really understand what
https://github.com/HypothesisWorks/hypothesis/blob/3fc1f2a76b0104d70b7a0800a8edcb01ad0c3dab/hypothesis-python/src/hypothesis/strategies/_internal/numbers.py#L111

Reply:
That's the `IntegerStrategy`'s implementation of the `filter()` method so the consumer can apply further filtering. It doesn't look as though it uses it itself.

Reply:
So, I've now updated this to how I _think_ the code should look for sorting/pick one row per patient.  But there is a weirdness that I've discovered (due to tedious and repeated trial and error, mostly):
With nothing cached, on a clean run of hypothesis, it always passes.  But on a second run, it always fails the health check.  (I found an old issue where someone was having unexpected health check failures, and David MacIver suggested trying to delete the `.hypothesis` folder and try again, although that did seem like a stab-in-the-dark suggestion)

Here's the hypothesis stats from a run with 100 examples that I just did, after deleting the .hypothesis folder:
```
============================= Hypothesis Statistics ==============================

tests/generative/test_data_setup.py::test_data_strategy_examples_round_trip:

  - during generate phase (0.05 seconds):
    - Typical runtimes: 3-44 ms, ~ 47% in data generation
    - 1 passing examples, 0 failing examples, 1 invalid examples

  - Stopped because settings.max_examples=1


tests/generative/test_query_model.py::test_query_model:

  - during generate phase (22.10 seconds):
    - Typical runtimes: 10-127 ms, ~ 84% in data generation
    - 100 passing examples, 0 failing examples, 448 invalid examples

  - Stopped because settings.max_examples=100
```

And then a second run, immediately afterwards:
```
==================================== FAILURES ====================================
________________________________ test_query_model ________________________________
Traceback (most recent call last):
  File "/home/becky/datalab/databuilder/tests/generative/test_query_model.py", line 85, in test_query_model
    @hyp.settings(**settings)
  File "/home/becky/datalab/databuilder/.venv/lib/python3.9/site-packages/hypothesis/core.py", line 1397, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/becky/datalab/databuilder/.venv/lib/python3.9/site-packages/hypothesis/internal/healthcheck.py", line 27, in fail_health_check
    raise FailedHealthCheck(message)
hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 0 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
----------------------------------- Hypothesis -----------------------------------
You can add @seed(44756034410180094038173309016381335108) to this test or run pytest with --hypothesis-seed=44756034410180094038173309016381335108 to reproduce this failure.
```

It's not always 0 out of 50, but it does always seem to fail the second time.

Reply:
It also will pass it I add `database=None` to the `@hyp.settings` decorator, but clearly we don't want to do that

Reply:
This seems weird enough that it's worth engaging with the Hypothesis community to try to understand it. As well as the Github repo there are some active experts on StackOverflow.